### PR TITLE
Refactor callbacks #CCM-48

### PR DESCRIPTION
**Pull request by literat**

### CODE REVIEW

**Review by OndraM — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -261,17 +261,14 @@ for example, to enable some feature after user has given consent.
...
 Each configured callback receives two params:
 
-* `cookie` - object with cookie contents
 * `cookieConsent` - instance of the underlying [cookie consent] library, can be used to call [its methods][cookie consent api]
 
 | Callback                     | Trigger event                                                                                                             |
 |------------------------------|---------------------------------------------------------------------------------------------------------------------------|
-| `onAcceptAll`                | Consent with all cookies is detected (either given now or after page load if it was already saved previously)             |
-| `onAcceptOnlyNecessary`      | Consent with only necessary cookies is detected (either given now or after page load if it was already saved previously)* |
+| `onFirstAction`           | Right after all cookies are accepted                                                                                      |
```
[the remark this answers is not in the document]
Comment (literat) [reply README.md]: Changed.

File: README.md [hunk 2]
```diff
@@ -261,17 +261,14 @@ for example, to enable some feature after user has given consent.
...
 | `onAccept`                   | Any consent is detected (either given now or after page load if it was already saved previously)                          |
-| `onFirstAcceptAll`           | Right after all cookies are accepted                                                                                      |
-| `onFirstAcceptOnlyNecessary` | Right after only necessary cookies are just accepted by the user                                                          |
-| `onFirstAccept`              | Right after any consent is just accepted by the user                                                                      |
+| `onChange` | Right after user change his/her preferences                                                          |
```
[the remark this answers is not in the document]
Comment (literat) [reply README.md]: Fixed.

File: src/types/CookieConsentManager.ts [hunk 3]
```diff
@@ -1,21 +1,28 @@
 import { CookieConsentCategory } from './CookieConsentCategory';
 import { VanillaCookieConsent } from './vanilla-cookieconsent';
 
-export type OnAcceptCallback = (
-  cookie: VanillaCookieConsent.Cookie<CookieConsentCategory>,
+type Callback = (cookieConsent: VanillaCookieConsent.CookieConsent<CookieConsentCategory>) => void;
```
[the remark this answers is not in the document]
Comment (literat) [reply src/types/CookieConsentManager.ts]: Developer's WET dreams are coming :D

[the remark this answers is not in the document]
Comment (literat) [reply src/types/CookieConsentManager.ts]: Did not know about this 👍  But in most cases I will stay with DRY ;-)

File: src/LmcCookieConsentManager.ts [hunk 4]
```diff
@@ -156,6 +158,9 @@ const LmcCookieConsentManager: CookieConsentManager = (serviceName, args) => {
   return cookieConsent;
 };
 
+/**
+ * @param cookie
+ */
```
[the remark this answers is not in the document]
Comment (literat) [reply src/LmcCookieConsentManager.ts:163]: Added somehow by IDE. I'll fix it.

**Review by OndraM — APPROVED** [review]

**Pull request merged**